src/libasr/pass/replace_symbolic.cpp

```diff
@@ -559,6 +559,84 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
+                    al, loc, current_scope, s2c(al, "args"), nullptr, 0, ASR::intentType::Local,
+                    nullptr, nullptr, ASR::storage_typeType::Default, CPtr_type, nullptr,
+                    ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, false));
+                //if (!current_scope->get_symbol("args")) {
+                current_scope->add_symbol(s2c(al, "args"), args_sym);
+                //}
+
+                // Statement 1
+                ASR::expr_t* args = ASRUtils::EXPR(ASR::make_Var_t(al, loc, args_sym));
+                Vec<ASR::call_arg_t> call_args1;
+                call_args1.reserve(al, 1);
+                ASR::expr_t* function_call1 = ASRUtils::EXPR(ASRUtils::make_FunctionCall_t_util(al, loc,
+                    vecbasic_new_sym, vecbasic_new_sym, call_args1.p, call_args1.n,
+                    ASRUtils::TYPE(ASR::make_CPtr_t(al, loc)), nullptr, nullptr));
+                ASR::stmt_t* stmt1 = ASRUtils::STMT(ASR::make_Assignment_t(al, loc, args, function_call1, nullptr));
+                pass_result.push_back(al, stmt1);
+
+                // Statement 2
+                Vec<ASR::call_arg_t> call_args2;
+                call_args2.reserve(al, 2);
+                ASR::call_arg_t call_arg1, call_arg2;
+                call_arg1.loc = loc;
+                call_arg1.m_value = value1;
+                call_arg2.loc = loc;
+                call_arg2.m_value = args;
+                call_args2.push_back(al, call_arg1);
+                call_args2.push_back(al, call_arg2);
+                ASR::stmt_t* stmt2 = ASRUtils::STMT(ASR::make_SubroutineCall_t(al, loc, basic_get_args_sym,
+                    basic_get_args_sym, call_args2.p, call_args2.n, nullptr));
+                pass_result.push_back(al, stmt2);
+
+                // Statement 3
+                Vec<ASR::stmt_t *> if_body; if_body.reserve(al, 1);
+                Vec<ASR::stmt_t *> else_body; else_body.reserve(al, 1);
+                Vec<ASR::call_arg_t> call_args3;
+                call_args3.reserve(al, 1);
+                ASR::call_arg_t call_arg3;
+                call_arg3.loc = loc;
+                call_arg3.m_value = args;
+                call_args3.push_back(al, call_arg3);
+                ASR::expr_t* function_call2 = ASRUtils::EXPR(ASRUtils::make_FunctionCall_t_util(al, loc,
+                    vecbasic_size_sym, vecbasic_size_sym, call_args3.p, call_args3.n,
+                    ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4)), nullptr, nullptr));
+                ASR::expr_t* int_compare = ASRUtils::EXPR(ASR::make_IntegerCompare_t(al, loc, function_call2, ASR::cmpopType::LtE,
+                        x->m_args[1], ASRUtils::TYPE(ASR::make_Logical_t(al, loc, 4)), nullptr));
+                ASR::stmt_t* stmt3 = ASRUtils::STMT(make_If_t(al, loc, int_compare,
+                    if_body.p, if_body.n, else_body.p, else_body.n));
+                pass_result.push_back(al, stmt3);
```

Comment:
Also this is the second thing which I wanted to point out. What I'm trying to do here is raise an error of some sort 
```
    if vecbasic_size(args) <= "the argument number passed":
        raise
```
I think I have the `test` ready but I need to raise an error of some sort to tell the user that the argument number queried for can't be returned or rather is not available. I am not fully sure how do I address the loopbody

Reply:
Currently the loop body is empty so I assume it's using `pass` or something.

Reply:
I've addressed this too as of now but I am somewhat doubtful on the implementation of `visit_Raise` in `python_ast_to_asr.cpp` file and also how the `ErrorStop` node works in the asr. So I've addressed this but I don't know how to I add a exception statement for the same. We have the following currently 
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main0():
    x: S = Symbol("x")
    x = x**pi
    arg1: S = x.args[1]
    print(arg1)

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --enable-symengine examples/expr2.py 
pi
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main0():
    x: S = Symbol("x")
    x = x**pi
    arg1: S = x.args[2]
    print(arg1)

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --enable-symengine examples/expr2.py 
ERROR STOP
```